velocity/src/main/java/eu/kennytv/maintenance/addon/velocity/listener/MessagingListener.java

```diff
@@ -14,6 +15,9 @@ public MessagingListener(final MaintenanceVelocityAddon plugin) {
 
     @Subscribe
     public void execute(final PluginMessageEvent event) {
+        if (!event.getIdentifier().getId().equals(MaintenanceChannel.REQUEST_CHANNEL_ID)) return;
```

Comment:
Now the identifier is checked twice, so this is not needed if you just move the rest down

Reply:
Ok that was an oversight I fixed it, the thing with the exploitable code was that the message would not be handled on the proxy but would be smuggled to the server which is not recommended